providers/flagsmith/src/main/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithProvider.java

```diff
@@ -0,0 +1,263 @@
+ */
+@Slf4j
+class FlagsmithProvider implements FeatureProvider {
+
+    private static final String NAME = "Flagsmith Provider";
+    private static FlagsmithClient flagsmith;
+    private FlagsmithProviderOptions options;
+
+    public FlagsmithProvider(FlagsmithProviderOptions options) {
+        this.options = options;
+        flagsmith = FlagsmithClientConfigurer.initializeProvider(options);
+    }
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> NAME;
+    }
+
+    @Override
+    public List<Hook> getProviderHooks() {
+        return FeatureProvider.super.getProviderHooks();
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(
+        String key, Boolean defaultValue, EvaluationContext evaluationContext) {
+        // When isUsingBooleanConfigValue the feature_state_value will be used as the flag value
+        if (this.options.isUsingBooleanConfigValue()) {
+            return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Boolean.class);
+        }
+        // Otherwise the enabled field will be used as the boolean flag value
+        try {
+            Flags flags = getFlags(evaluationContext);
+            boolean isFlagEnabled = flags.isFeatureEnabled(key);
+            Reason reason = isFlagEnabled ? null : Reason.DISABLED;
+            return buildEvaluation(isFlagEnabled, null, reason, null);
+        } catch (FlagsmithClientError flagsmithClientError) {
+            return buildEvaluation(defaultValue, ErrorCode.GENERAL, Reason.ERROR, null);
+        }
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(
+        String key, String defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, String.class);
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(
+        String key, Integer defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Integer.class);
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(
+        String key, Double defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Double.class);
+    }
+
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(
+        String key, Value defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Value.class);
+    }
+
+    /**
+     * Get all the flags for a given environment or identity. The Method will use
+     * getEnvironmentFlags from the Flagsmith sdk if no targeting key is provided
+     * in the EvaluationContext. If a targeting key is provided then the
+     * getIdentityFlags method will be used.
```

Comment:
I think this is fine, but you should mention this in your doc IMHO.

You also _may_ want to warn if a context attr is set without setting a targeting key, since it would have not effect in this case if I understand correctly.

Reply:
I've added a description on how to do this in the readme